centreon/src/Core/Service/Application/Model/NotificationTypeConverter.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Service\Application\Model;
+
+use Assert\AssertionFailedException;
+use Centreon\Domain\Common\Assertion\Assertion;
+use Core\Service\Domain\Model\NotificationType;
+
+class NotificationTypeConverter
+{
+    public const NONE_AS_BIT = 0b000000,
+                 WARNING_AS_BIT = 0b000001,
+                 UNKNOWN_AS_BIT = 0b000010,
+                 CRITICAL_AS_BIT = 0b000100,
+                 RECOVERY_AS_BIT = 0b001000,
+                 FLAPPING_AS_BIT = 0b010000,
+                 DOWNTIME_SCHEDULED_AS_BIT = 0b100000,
+                 ALL_TYPE = 0b111111;
+
+    /**
+     * @param int $bitFlag
+     *
+     * @throws AssertionFailedException
+     *
+     * @return list<NotificationType>
+     */
+    public static function fromBits(int $bitFlag): array
+    {
+        Assertion::range($bitFlag, 0, self::ALL_TYPE);
+        if ($bitFlag === self::NONE_AS_BIT) {
+            return [NotificationType::None];
+        }
+
+        $notificationTypes = [];
+        foreach (NotificationType::cases() as $notificationType) {
+            if ($bitFlag & self::toBit($notificationType)) {
+                $notificationTypes[] = $notificationType;
+            }
+        }
+
+        return $notificationTypes;
+    }
+
+    /**
+     * @param NotificationType $notificationType
+     *
+     * @return int
+     */
+    private static function toBit(NotificationType $notificationType): int
```

Comment:
```suggestion
    public static function toBit(NotificationType $notificationType): int
```
better to set it as public as we could use it outside of the class itself